read-fonts/src/tables/cmap.rs

```diff
@@ -91,6 +102,57 @@ impl<'a> Cmap12<'a> {
     }
 }
 
+impl<'a> Cmap14<'a> {
+    /// Maps a codepoint and variation selector to a nominal glyph identifier.
+    pub fn map_variant(
+        &self,
+        codepoint: impl Into<u32>,
+        selector: impl Into<u32>,
+    ) -> Option<MapVariant> {
+        let codepoint = codepoint.into();
+        let selector = selector.into();
+        let selector_records = self.var_selector();
+        // Variation selector records are sorted in order of var_selector. Binary search to find
+        // the appropriate record.
+        let selector_record = match selector_records
+            .binary_search_by(|record| <_ as Into<u32>>::into(record.var_selector()).cmp(&selector))
+        {
+            Ok(idx) => selector_records.get(idx)?,
+            _ => return None,
+        };
+        // If a default UVS table is present in this selector record, binary search on the ranges
+        // (start_unicode_value, start_unicode_value + additional_count) to find the requested codepoint.
+        // If found, ignore the selector and return a value indicating that the default cmap mapping
+        // should be used.
+        if let Some(Ok(default_uvs)) = selector_record.default_uvs(self.offset_data()) {
+            let ranges = default_uvs.ranges();
+            let mut lo = 0;
+            let mut hi = ranges.len();
+            while lo < hi {
+                let i = (lo + hi) / 2;
+                let range = &ranges[i];
+                let start = range.start_unicode_value().into();
+                if codepoint < start {
+                    hi = i;
+                } else if codepoint > (start + range.additional_count() as u32) {
+                    lo = i + 1;
+                } else {
+                    return Some(MapVariant::UseDefault);
+                }
+            }
```

Comment:
Why are you using binary_search_by below but not above? - when I tried it, convinced there was some problem with the idea I hadn't spotted, it passed the tests and the code felt better:

```rust
if default_uvs.ranges().binary_search_by(|range| {
    let start = range.start_unicode_value().into();
    if codepoint < start {
        Ordering::Greater
    } else if codepoint > (start + range.additional_count() as u32) {
        Ordering::Less
    } else {
        Ordering::Equal
    }
}).is_ok() {
    return Some(MapVariant::UseDefault);
}
```